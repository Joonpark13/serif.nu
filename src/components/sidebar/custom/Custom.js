import React, { useState, Fragment } from 'react';
import { useDispatch } from 'react-redux';
import { Button } from '@material-ui/core';
import TextField from '@material-ui/core/TextField';
import FormControl from '@material-ui/core/FormControl';
import FormGroup from '@material-ui/core/FormGroup';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import FormHelperText from '@material-ui/core/FormHelperText';
import Checkbox from '@material-ui/core/Checkbox';
import { makeStyles } from '@material-ui/styles';
import { addSectionFromCustom } from 'actions';

const useStyles = makeStyles({
  container: {
    marginLeft: 15,
    marginRight: 15,
    display: 'flex',
  },
  time: {
    marginRight: 15,
  },
});


export default function Custom() {
  const classes = useStyles();
  const [title, changeTitle] = useState('');
  const [description, changeDescription] = useState('');
  const [startTime, changeStartTime] = useState('9:00');
  const [endTime, changeEndTime] = useState('9:30');
  const [eventDow, setEventDow] = useState({
    monday: false,
    tuesday: false,
    wednesday: false,
    thursday: false,
    friday: false,
  });

  const canSubmit = !!title
    && !!description
    && !!startTime
    && !!endTime
    && Object.values(eventDow).some(dow => dow);

  const handleTextFieldChange = changeFunc => (event) => { changeFunc(event.target.value); };
  const handleChange = name => (event) => {
    setEventDow({ ...eventDow, [name]: event.target.checked });
  };

  const { monday, tuesday, wednesday, thursday, friday } = eventDow;
  const error = [monday, tuesday, wednesday, thursday, friday].filter(v => v).length < 1;

  const dowMap = ['Mo', 'Tue', 'Wed', 'Thurs', 'Fri'];
  const dispatch = useDispatch();
  const handleSubmit = () => {
    const dow = Object.values(eventDow).map((v, i) => v ? dowMap[i] : null).filter(v => v);
    const [startHour, startMinute] = startTime.split(':').map(x => parseInt(x, 10));
    const [endHour, endMinute] = endTime.split(':').map(x => parseInt(x, 10));
    const schedule = {
      dow,
      start: {
        hour: startHour,
        minute: startMinute,
      },
      end: {
        hour: endHour,
        minute: endMinute,
      },
      location: '',
    };
    const section = {
      id: (Math.random() * 100).toString(),
      termId: '',
      schoolId: '',
      subjectId: '',
      courseId: '',
      name: title,
      sectionNumber: '',
      topic: '',
      descriptions: [{ name: 'Description', value: description }],
      instructors: [''],
      schedules: [schedule],
    };
    dispatch(addSectionFromCustom(section));
  };

  return (
    <Fragment>
      <div className={classes.container}>
        <FormControl required error={error} component="fieldset" fullWidth margin="normal">
          <TextField
            required
            id="standard-required"
            label="Class Title"
            margin="normal"
            onChange={handleTextFieldChange(changeTitle)}
          />
          <TextField
            required
            id="standard-required"
            label="Description"
            margin="normal"
            onChange={handleTextFieldChange(changeDescription)}
          />
          <FormGroup>
            <FormHelperText component="legend">Occurrences</FormHelperText>
            <FormControlLabel
              control={<Checkbox checked={monday} onChange={handleChange('monday')} value="monday" />}
              label="Monday"
            />
            <FormControlLabel
              control={<Checkbox checked={tuesday} onChange={handleChange('tuesday')} value="tuesday" />}
              label="Tuesday"
            />
            <FormControlLabel
              control={<Checkbox checked={wednesday} onChange={handleChange('wednesday')} value="wednesday" />}
              label="Wednesday"
            />
            <FormControlLabel
              control={<Checkbox checked={thursday} onChange={handleChange('thursday')} value="thursday" />}
              label="Thursday"
            />
            <FormControlLabel
              control={<Checkbox checked={friday} onChange={handleChange('friday')} value="friday" />}
              label="Friday"
            />
          </FormGroup>
          <FormHelperText>At least one</FormHelperText>
          <div>
            <TextField
              className={classes.time}
              id="time"
              label="Start Time"
              type="time"
              margin="normal"
              defaultValue="09:00"
              InputLabelProps={{
                shrink: true,
              }}
              inputProps={{
                step: 300, // 5 min
              }}
              onChange={handleTextFieldChange(changeStartTime)}
            />
            <TextField
              className={classes.time}
              id="time"
              label="End Time"
              type="time"
              margin="normal"
              defaultValue="09:30"
              InputLabelProps={{
                shrink: true,
              }}
              inputProps={{
                step: 300, // 5 min
              }}
              onChange={handleTextFieldChange(changeEndTime)}
            />
          </div>
          <Button onClick={handleSubmit} disabled={!canSubmit}>Create</Button>
        </FormControl>
      </div>
    </Fragment>
  );
}
