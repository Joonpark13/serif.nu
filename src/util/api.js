/* eslint-disable import/prefer-default-export */
export function searchURL(query) {
  return `${process.env.API_URL}/search?term_id=4720&search_query=${query}`;
}

export function sectionsURL(termId, schoolAbbv, subjectAbbv, courseAbbv) {
  return `${process.env.API_URL}/sections?term_id=${termId}&school_abbv=${schoolAbbv}&subject_abbv=${subjectAbbv}&course_abbv=${courseAbbv}`;
}
