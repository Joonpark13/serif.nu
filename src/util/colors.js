export const northwesternPurple = '#4E2A84';
export const northwesternBrightOrange = '#FFC520';
export const northwesternRichBlack10 = '#D8D6D6';
