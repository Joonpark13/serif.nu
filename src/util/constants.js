/* eslint-disable import/prefer-default-export */
export const CURRENT_TERM = '4730';
