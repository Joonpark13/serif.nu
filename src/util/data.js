import termsData from 'data/terms.json';

/* eslint-disable import/prefer-default-export */
/* istanbul ignore next */
export const currentTerm = termsData.reduce((highestTerm, term) => {
  if (term.id > highestTerm.id) {
    return term;
  }
  return highestTerm;
});
/* eslint-enable import/prefer-default-export */
