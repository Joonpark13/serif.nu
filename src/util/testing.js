/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import * as reactRedux from 'react-redux';
import { shallow } from 'enzyme';
import * as useSelector from './use-selector';

jest.mock('./use-selector');

// Helper function that returns a function which creates enzyme wrappers with default props
export function wrapperCreator(Component, defaultProps = {}) {
  let classes;
  return props => shallow(<Component classes={classes} {...defaultProps} {...props} />);
}

/* istanbul ignore next */
export function mockUseSelector(...args) {
  useSelector.default.mockReset();
  args.forEach(arg => useSelector.default.mockReturnValueOnce(arg));

  return useSelector.default;
}

/* istanbul ignore next */
export function mockUseDispatch() {
  const dispatchSpy = jest.fn();
  reactRedux.useDispatch.mockReturnValue(dispatchSpy);
  return dispatchSpy;
}
