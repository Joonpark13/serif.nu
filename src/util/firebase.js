import firebase from 'firebase/app';
import 'firebase/firestore';
import 'firebase/auth';

firebase.initializeApp({
  apiKey: 'AIzaSyDluAG6bnqleqJ3yjPvGdlb3RyZP3U5JIQ',
  authDomain: 'sans-serif-northwestern.firebaseapp.com',
  databaseURL: 'https://sans-serif-northwestern.firebaseio.com',
  projectId: 'sans-serif-northwestern',
  storageBucket: 'sans-serif-northwestern.appspot.com',
  messagingSenderId: '1078100635889',
});

export const db = firebase.firestore();
export const auth = firebase.auth();
