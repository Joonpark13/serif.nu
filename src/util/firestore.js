import firebase from 'firebase/app';
import 'firebase/firestore';

firebase.initializeApp({
  apiKey: 'AIzaSyDluAG6bnqleqJ3yjPvGdlb3RyZP3U5JIQ',
  authDomain: 'sans-serif-northwestern.firebaseapp.com',
  databaseURL: 'https://sans-serif-northwestern.firebaseio.com',
  projectId: 'sans-serif-northwestern',
  storageBucket: 'sans-serif-northwestern.appspot.com',
  messagingSenderId: '1078100635889',
});

export default firebase.firestore();
