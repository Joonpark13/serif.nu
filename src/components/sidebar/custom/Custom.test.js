import React from 'react';
import { shallow } from 'enzyme';
import { mockUseSelector } from 'util/testing';
import Custom from './Custom';

describe('Custom', () => {
  it('renders correctly', () => {
    mockUseSelector('custom');

    const wrapper = shallow(<Custom />);

    expect(wrapper.get(0)).toMatchSnapshot();
  });
});
