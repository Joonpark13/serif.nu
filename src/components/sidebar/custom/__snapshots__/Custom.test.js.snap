// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom renders correctly 1`] = `
<React.Fragment>
  <div
    className="makeStyles-container-1"
  >
    <ForwardRef(WithStyles)
      component="fieldset"
      error={true}
      fullWidth={true}
      margin="normal"
      required={true}
    >
      <ForwardRef(WithStyles)
        id="standard-required"
        label="Class Title"
        margin="normal"
        onChange={[Function]}
        required={true}
      />
      <ForwardRef(WithStyles)
        id="standard-required"
        label="Description"
        margin="normal"
        onChange={[Function]}
        required={true}
      />
      <ForwardRef(WithStyles)>
        <ForwardRef(WithStyles)
          component="legend"
        >
          Occurrences
        </ForwardRef(WithStyles)>
        <ForwardRef(WithStyles)
          control={
            <ForwardRef(WithStyles)
              checked={false}
              onChange={[Function]}
              value="monday"
            />
          }
          label="Monday"
        />
        <ForwardRef(WithStyles)
          control={
            <ForwardRef(WithStyles)
              checked={false}
              onChange={[Function]}
              value="tuesday"
            />
          }
          label="Tuesday"
        />
        <ForwardRef(WithStyles)
          control={
            <ForwardRef(WithStyles)
              checked={false}
              onChange={[Function]}
              value="wednesday"
            />
          }
          label="Wednesday"
        />
        <ForwardRef(WithStyles)
          control={
            <ForwardRef(WithStyles)
              checked={false}
              onChange={[Function]}
              value="thursday"
            />
          }
          label="Thursday"
        />
        <ForwardRef(WithStyles)
          control={
            <ForwardRef(WithStyles)
              checked={false}
              onChange={[Function]}
              value="friday"
            />
          }
          label="Friday"
        />
      </ForwardRef(WithStyles)>
      <ForwardRef(WithStyles)>
        At least one
      </ForwardRef(WithStyles)>
      <div>
        <ForwardRef(WithStyles)
          InputLabelProps={
            Object {
              "shrink": true,
            }
          }
          className="makeStyles-time-2"
          defaultValue="09:00"
          id="time"
          inputProps={
            Object {
              "step": 300,
            }
          }
          label="Start Time"
          margin="normal"
          onChange={[Function]}
          type="time"
        />
        <ForwardRef(WithStyles)
          InputLabelProps={
            Object {
              "shrink": true,
            }
          }
          className="makeStyles-time-2"
          defaultValue="09:30"
          id="time"
          inputProps={
            Object {
              "step": 300,
            }
          }
          label="End Time"
          margin="normal"
          onChange={[Function]}
          type="time"
        />
      </div>
      <ForwardRef(WithStyles)
        disabled={true}
        onClick={[Function]}
      >
        Create
      </ForwardRef(WithStyles)>
    </ForwardRef(WithStyles)>
  </div>
</React.Fragment>
`;
